### PR TITLE
SALTO-6376: Adjust Workato config options type

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Adjust Workato config object type 

---

_Additional context for reviewer_
We allow working with Workato in 2 modes - deploy support or impact analysis. each have implications on the adapter config, nacls structure, and more.
The object type should reflect this selection.

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/workato-adapter/src/config_creator.ts [hunk 1]
```diff
@@ -25,13 +32,42 @@ import { WORKATO } from './constants'
...
+    useCase: {
+      refType: BuiltinTypes.STRING,
+      annotations: {
+        [CORE_ANNOTATIONS.DEFAULT]: WORKATO_DEPLOY_OPTION,
+        [CORE_ANNOTATIONS.REQUIRED]: true,
+        [CORE_ANNOTATIONS.ALIAS]: 'Choose your Workato use case',
+        [CORE_ANNOTATIONS.RESTRICTION]: createRestriction({
+          values: [WORKATO_DEPLOY_OPTION, WORKATO_IMPACT_ANALYSIS_OPTION],
+          enforce_value: true,
+          max_list_length: 1,
```
Comment (adfineberg) [inline packages/workato-adapter/src/config_creator.ts:54]: why do you need this? it's a string restriction, not a list one

File: packages/workato-adapter/src/config_creator.ts [hunk 2]
```diff
@@ -40,9 +76,10 @@ export const getConfig = async (options?: InstanceElement): Promise<InstanceElem
   if (options === undefined || !createOptionsTypeGuard<ConfigOptionsType>(optionsElemId)(options)) {
     return defaultConfig
   }
-  if (options.value.enableDeploy !== undefined) {
```
Comment (adfineberg) [inline packages/workato-adapter/src/config_creator.ts:43]: Do you want to have a check for this for backwards compatibility? If someone doesn't refresh after deploy and wants to connect a new Workato env. Yeah, sounds kind of farfetched...

**Review by adfineberg — APPROVED** [review]
Looks good! A couple of small comments, pre-approving to save cycles

File: packages/workato-adapter/src/config_creator.ts (hunk 2 above)
Comment (shir-reifenberg) [reply packages/workato-adapter/src/config_creator.ts:43]: discussed IRL

File: packages/workato-adapter/src/config_creator.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/workato-adapter/src/config_creator.ts:54]: leftover 😅

**Pull request merged**